### PR TITLE
Use cythonize to simplify install script

### PR DESCRIPTION
**Pull request by okuta**
This PR provides simple install script.
This PR includes #1775, #1850, #1854 and #1850.

### CODE REVIEW

**Comment by gwtnb** [comment]
Please describe difference from #1775.

**Comment by okuta** [comment]
This PR includes #1775, #1850, #1854 and #1850.

File: chainer_setup_build.py [hunk 1]
```diff
@@ -1,23 +1,19 @@
...
 import setuptools
-from setuptools.command import build_ext
 
 from install import build
 from install import utils
 
 
-dummy_extension = setuptools.Extension('chainer', ['chainer.c'])
-cython_version = '0.23.0'
+require_cython_version = pkg_resources.parse_version('0.24.0')
```
Comment (gwtnb) [inline chainer_setup_build.py]: s/require/required/

File: chainer_setup_build.py [hunk 2]
```diff
@@ -193,125 +159,83 @@ def make_extensions(options, compiler):
...
 
-def to_cpp_extensions(extensions):
-    ret = []
-    for x in extensions:
-        ext = copy.copy(x)
-        ext.sources = [path.splitext(f)[0] + ".cpp" for f in x.sources]
-        ret.append(ext)
-    return ret
+    return Cython.Build.cythonize(
+        extensions, language="c++", verbose=True,
```
Comment (gwtnb) [inline chainer_setup_build.py]: `language` argument is deprecated. https://github.com/cython/cython/blob/f93e8de11de0f6e7236d27fad1806104f354ad77/Cython/Build/Dependencies.py#L638

File: chainer_setup_build.py [hunk 3]
```diff
@@ -193,125 +159,83 @@ def make_extensions(options, compiler):
...
 
 
-_arg_options = {}
-
-
 def parse_args():
-    global _arg_options
-    _arg_options['profile'] = '--cupy-profile' in sys.argv
-    if _arg_options['profile']:
+    arg_options = dict()
```
Comment (gwtnb) [inline chainer_setup_build.py]: `{}` is used on Chainer.

**Comment by okuta** [comment]
When will this PR be merged?

**Comment by gwtnb** [comment]
LGTM

**Pull request merged**